### DISCUSSION TITLE
Issues with custom client development: npm run dev / local hangs on ZipLoader.module.js

### DISCUSSION THREAD

**Discussion by misslivirose** [mozilla/hubs#3147] [Hubs Cloud Questions] [answered]
I am working on an Ubuntu 20.04.1 LTS virtual machine (via Hyper-V) to try and build a custom client that I deploy to a Hubs Cloud instance. However, I am unable to get through the `npm run dev` or `npm run local` scripts after running `npm ci` because the script appears to hang after printing the following: 

```
assets/js/sketchfab-zip.worker-abf8e2dcc563a1a63258.js 230 KiB main [emitted][immutable] [bit] main
Entrypoint main = assets/js/sketchfab-zip.worker-abf8e2dcc563a1a63258.js] ./node_modeules/babel-loader/lib!./src/workers/sketchfab-zip.worker.js 2 KiB {main} [built]
[./node_modules/zip-loader/dist/ZipLoader.module.js] 117 KiB {main} [built]
[wdm]: Compiled successfully.
```

When I try to visit https://localhost:8080, I get a 'Connection timed out' error.

I'll keep the VM up to see if it's possible it's just taking a _really_ long time (20+ minutes here) to deploy, but also wanted to document this as a discussion in case others have this issue and have suggestions.

**Comment by johnshaughnessy** [comment]
I may be missing some context here, but I don't think `npm run dev/local` are necessary. The docs suggest: 
- `npm run login` to authenticate with your hubs cloud instance
- `npm start` to setup local testing
- Test custom client
- `npm run deploy` - to deploy
 
More information / troubleshooting help can be found in ["Creating and Deploying Custom Clients"](https://hubs.mozilla.com/docs/hubs-cloud-custom-clients.html) section of the docs. If those steps aren't working or if they are unclear, any information you can provide may help us improve them.

**Reply by misslivirose** [reply]
Ah, thank you! I was following the instructions at the top of the readme Quick Start: 

```
git clone https://github.com/mozilla/hubs.git
cd hubs
npm ci
npm run dev
```
I'll try again with the other commands that you suggested. I have authenticated, and so far running `npm start` seems to be doing the same thing, hanging on the same location, but I'll let it run for a while and see if it resolves this time.

**Reply by robinkwilson** [reply]
You could also try deleting the **node_modules** folder and run `npm ci` again so you have a fresh start.

**Reply by misslivirose** [reply]
Thanks Robin - same issue after deleting the folder and re-running `npm ci`.

**Reply by johnshaughnessy** [reply]
I misunderstood the first error. The message you received in the console is correct, and means that hubs built successfully. All of these commands `npm run start/dev/local` will build the js bundle and then start a local web server for you to test on. In other words, it's normal for the console to be "idling" after you run one of those commands, and it will print out messages as you connect to the web server through your browser.

It sounds like the issue has more to do with not being able to connect to the local web server. Are you connecting to `https://localhost:8080` in the VM itself? I have not worked with the same setup, but I'm wondering if it has something to do with the network configuration of the VM.

**Comment by misslivirose** [comment, marked as answer]
Update: I had an old version of node installed. Updating to the latest version of node fixed this. If anyone else is having the issue where responses time out, check your version of node with `node --version` and update if it's behind. I was on version 10, which is what ships with Ubuntu 20 by default.
